### ISSUE TITLE
Configure javadoc plugin to make use of compileOnly scope

### ISSUE THREAD

**Issue by jbgi** [tbroyer/gradle-apt-plugin#5]
Currenlty javadoc emmits warnings. eg:
```
error: package com.google.auto.service does not exist
@com.google.auto.service.AutoService(Processor.class)
                        ^
javadoc: warning - Class com.google.auto.service.AutoService not found.
```

**Comment by jbgi** [comment]
And thanks for this great plugin!

**Comment by tbroyer** [comment]
It's actually a timing issue, with the `javadoc` task picking its classpath before we reconfigure it. Should be fixed by fa4c0565c11ce6f07f57a2a1e86842c8dac25a18. Will make a release soon.

**Comment by jbgi** [comment]
great! thanks!

**Comment by tbroyer** [comment]
Just published v0.4

**Comment by jbgi** [comment]
all is good now! thanks.
